src/braket/pulse/pulse_sequence.py

```diff
@@ -68,6 +68,21 @@ def parameters(self) -> Set[FreeParameter]:
         """Returns the set of `FreeParameter`s in the PulseSequence."""
         return self._free_parameters.copy()
 
+    @property
+    def waveforms(self):
+        """Returns the map of `Waveform` IDs to `Waveform`s in the PulseSequence."""
+        return self._waveforms
+
+    @property
+    def frames(self):
+        """Returns the map of `Frame` IDs to `Frame`s in the PulseSequence."""
+        return self._frames
+
+    @property
+    def program(self):
+        """Returns the `Program` in the PulseSequence."""
+        return self._program
```

Comment:
What's our rationale for exposing these oqpy data structures directly?

Reply:
This is something that would not be needed and upon further consideration may actually cause confusion. We mutate this prop in `to_ir` such that `self._program.to_qasm() != self.to_ir()`